File: sdk/go/synaptic_test.go
package heartlight

import (
	"math"
	"testing"
)

func TestSynapticConformanceV1(t *testing.T) {
	state := SynapticState{Weight: 0.2, Eligibility: 0.0}
	inputs := []SynapticInput{{0.5, 0.25, 1.0}, {1.0, -0.5, 0.4}, {0.2, 0.8, -0.25}}
	weights := []float64{0.2096, 0.1967808, 0.2001622384}
	traces := []float64{0.125, -0.3875, -0.18875}
	for i, input := range inputs {
		state = SynapticStep(state, input, DefaultSynapticConfig())
		if math.Abs(state.Weight-weights[i]) >= 1e-12 || math.Abs(state.Eligibility-traces[i]) >= 1e-12 {
			t.Fatalf("step %d mismatch: %+v", i, state)
		}
	}
}

File: sdk/typescript/test/synaptic.test.mjs
import assert from "node:assert/strict";
import { defaultSynapticConfig, synapticStep } from "../dist/index.js";

let state = { weight: 0.2, eligibility: 0.0 };
const inputs = [
  { pre: 0.5, post: 0.25, reward: 1.0 },
  { pre: 1.0, post: -0.5, reward: 0.4 },
  { pre: 0.2, post: 0.8, reward: -0.25 },
];
const expected = [
  { weight: 0.2096, eligibility: 0.125 },
  { weight: 0.1967808, eligibility: -0.3875 },
  { weight: 0.2001622384, eligibility: -0.18875 },
];
for (let i = 0; i < inputs.length; i += 1) {
  state = synapticStep(state, inputs[i], defaultSynapticConfig());
  assert.ok(Math.abs(state.weight - expected[i].weight) < 1e-12);
  assert.ok(Math.abs(state.eligibility - expected[i].eligibility) < 1e-12);
}
